File: cognitodemo/signUp.js
const Cognito = require("@aws-sdk/client-cognito-identity-provider");
const crypto = require("crypto");

const clientId = "";  // Obtain from the AWS console
const clientSecret = "";  // Obtain from the AWS console
const username = "theuser";
const password = "supersecret";
const email = "your email address goes here";



function secretHash(clientId, clientSecret, username) {
  const hasher = crypto.createHmac('sha256', clientSecret);
  hasher.update(`${username}${clientId}`);
  return hasher.digest('base64');
}

async function main() {
  console.log("Signing up user");
  const client = new Cognito.CognitoIdentityProviderClient({ region: 'ap-southeast-2' });
  const command = new Cognito.SignUpCommand({
    ClientId: clientId,
    SecretHash: secretHash(clientId, clientSecret, username),
    Username: username,
    Password: password,
    UserAttributes: [{ Name: "email", Value: email }],
  });
  const res = await client.send(command);
  console.log(res);
}

main();

File: cognito_python/signUp.py
import boto3
import hmac, hashlib, base64 

username= "testuser"
password= "MySuperSecret99!"
email= "Your email goes here"
client_id = ""  # Obtain from the AWS console
client_secret = ""  # Obtain from the AWS console

def secretHash(clientId, clientSecret, username):
    message = bytes(username + clientId,'utf-8') 
    key = bytes(clientSecret,'utf-8') 
    return base64.b64encode(hmac.new(key, message, digestmod=hashlib.sha256).digest()).decode() 

def signup():
    client = boto3.client("cognito-idp", region_name="ap-southeast-2")
    try:
        response = client.sign_up(
            ClientId=client_id,
            Username=username,
            Password=password,
            SecretHash=secretHash(client_id, client_secret, username),
            UserAttributes=[{"Name": "email", "Value": email}]
        )
        return response
    except Exception as e:
        print(f"Error during sign-up: {e}")
        return None


signup_response = signup()
print("Sign-up successful:", signup_response)
